service/s3/s3crypto/s3crypto_client.go

```diff
@@ -0,0 +1,217 @@
+package s3crypto
+
+// DR TOOLS TEST integration test bucket
+import (
+	"bytes"
+	"encoding/base64"
+	"io/ioutil"
+	"math/rand"
+
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+const (
+	ivSize  = 16
+	keySize = 32
+)
+
+// Client supports client level encryption and decryption to S3.
+// The crypto client uses "envelope" encryption".
+type Client struct {
+	s3client *s3.S3
+	cfg      Config
+}
+
+// Config used to customize the Client
+type Config struct {
+	// MasterCipher will hold either a master symmetric key or a KMS client
+	// for encrypting the envelope key
+	MasterCipher Cipher
+}
+
+// NewClient will return a crypto client and set an aes cbc encrypter
+// as one of the fields.
+func NewClient(p client.ConfigProvider, options ...func(*Config)) (*Client, error) {
```

Comment:
Probably can go with just `New` instead of `NewClient`, but debatable because other types within this package.